File: hw/pastor/u1/hw14SafeInJsonChickenFarmSimulator/Chicken Farm Simulator (JS)/ec.espe.edu.chickenFarmSimulator.model/chickenFarmer.js
const ChickenCoop = require("./chickenCoops.js");

class ChickenFarmer {
    constructor(id, name) {
        this._id = id;
        this._name = name;
        this._coops = [];
        this._nextCoopId = 1;
    }

    addCoop(chickenCapacity) {
        const newCoop = new ChickenCoop(this._nextCoopId++, chickenCapacity);
        this._coops.push(newCoop);
        return newCoop;
    }

    removeCoop(coopId) {
        const initialLength = this._coops.length;
        this._coops = this._coops.filter(coop => coop.getCoopCoopNumber() !== coopId);
        return this._coops.length < initialLength;
    }

    updateCoop(coopId, coopUpdates) {
        const coop = this.findCoop(coopId);
        if (coop) {
            if (coopUpdates.capacity !== undefined) {
                coop.setCapacity(coopUpdates.capacity);
            }
            return true;
        }
        return false;
    }

    findCoop(coopId) {
        return this.getCoops().find(c => c.getCoopCoopNumber() === coopId);
    }

    toString() {
        return `Farmer: ${this.getName()} (ID: ${this.getId()}), Chicken coops: ${this.getCoops().length}`;
    }

    getId() { return this._id; }
    getName() { return this._name; }
    getCoops() { return this._coops; }
    getNextCoopId() { return this._nextCoopId; }

    setId(id) { this._id = id; }
    setName(name) { this._name = name; }
    setNextCoopId(nextCoopId) { this._nextCoopId = nextCoopId; }
}

module.exports = ChickenFarmer;

File: hw/pastor/u1/hw14SafeInJsonChickenFarmSimulator/Chicken Farm Simulator (PY)/espe_edu_ec_chicken_farm_simulator_model/chicken_farmer.py
from .chicken_coops import ChickenCoop

class ChickenFarmer:
    def __init__(self, id, name):
        self._id = id
        self._name = name
        self._coops = []
        self._next_coop_id = 1

    def add_coop(self, chicken_capacity):
        new_coop = ChickenCoop(self._next_coop_id, chicken_capacity)
        self._next_coop_id += 1
        self._coops.append(new_coop)
        return new_coop

    def remove_coop(self, coop_id):
        initial_length = len(self._coops)
        self._coops = [coop for coop in self._coops if coop.get_coop_coop_number() != coop_id]
        return len(self._coops) < initial_length

    def update_coop(self, coop_id, coop_updates):
        coop = self.find_coop(coop_id)
        if coop:
            if coop_updates.get('capacity') is not None:
                coop.set_capacity(coop_updates['capacity'])
            return True
        return False

    def find_coop(self, coop_id):
        for c in self._coops:
            if c.get_coop_coop_number() == coop_id:
                return c
        return None

    def __str__(self):
        return f"Farmer: {self.get_name()} (ID: {self.get_id()}), Chicken coops: {len(self.get_coops())}"

    def get_id(self): return self._id
    def get_name(self): return self._name
    def get_coops(self): return self._coops
    def get_next_coop_id(self): return self._next_coop_id

    def set_id(self, id): self._id = id
    def set_name(self, name): self._name = name
    def set_next_coop_id(self, next_coop_id): self._next_coop_id = next_coop_id